### PR TITLE
Implement JWT deserialization by keyset

### PR DESCRIPTION
**Pull request by simo5**
This allows passing of a keyset, and JWT will try to find the correct
key from the kid value in the header, and then decrypt or verify with
the correct key.

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>
Signed-off-by: Simo Sorce <simo@redhat.com>

### CODE REVIEW

**Comment by simo5** [comment]
This PR extends and replaces #23

**Comment by simo5** [comment]
@puiterwijk the above is just a slight variation on your code and retains your commit, can you take a look and see if it works for you ?

File: jwcrypto/jwt.py [hunk 1]
```diff
@@ -131,9 +168,9 @@ def __init__(self, header=None, claims=None, jwt=None, key=None,
          present in the token, if the value is not None the claim must
          match exactly.
 
-        Note: either the header,claims or jwt,key parameters should be
-        provided as a deserialization operation (which occurs if the jwt
-        is provided will wipe any header os claim provided by setting
+        Note: either the header,claims or jwt,key or jwt,keyset parameters
```
Comment (puiterwijk) [inline jwcrypto/jwt.py]: This still references the keyset parameter.

**Comment by puiterwijk** [comment]
Other than documentation, this looks good to me.

File: jwcrypto/jwt.py (hunk 1 above)
Comment (simo5) [inline jwcrypto/jwt.py]: good catch

**Comment by puiterwijk** [comment]
ack

**Pull request closed without merging**